viamonvif/service.go

```diff
@@ -103,15 +115,17 @@ func (dis *rtspDiscovery) DiscoverResources(ctx context.Context, extra map[strin
 	if len(list.Cameras) == 0 {
 		return nil, errors.New("no cameras found, ensure cameras are working or check credentials")
 	}
-
+	// clear dis.URIs list
+	dis.URIs = []URI{}
 	for _, camInfo := range list.Cameras {
 		dis.logger.Debugf("%s %s %s", camInfo.Manufacturer, camInfo.Model, camInfo.SerialNumber)
 		// some cameras return with no urls. explicitly skipping those so the behavior is clear in the service.
-		if len(camInfo.RTSPURLs) == 0 {
+		if len(camInfo.URIs) == 0 {
 			dis.logger.Errorf("No urls found for camera, skipping. %s %s %s",
 				camInfo.Manufacturer, camInfo.Model, camInfo.SerialNumber)
 			continue
 		}
+		dis.URIs = append(dis.URIs, camInfo.URIs...)
```

Comment:
TODO(seanp): Since we are manipulated struct level list URIs, from an API endpoint, we need to add a locking mechanism.